[WIP] Implement `odo dev --no-commands` (#6855)
redhat-developer/odo: docs/website/docs/command-reference/dev.md

```
</details>

### Running with no commands

The `--no-commands` flag allows to start the Dev Session without implicitly executing any `build`, `run` or `debug` commands.

Specifying this flag will simply create all the resources necessary for the Dev Session, then set up file synchronization and port forwarding.
In detail, it will:
1. eventually build and push any `Image` Components defined in the Devfile, if they have the `autoBuild` flag set to `true`
2. eventually apply any `Kubernetes` or `OpenShift` Components defined in the Devfile, if they have the `deployByDefault` flag set to `true`
3. Start the Dev container(s), taking care of executing any commands that are bound to [Devfile lifecycle events](../user-guides/advanced/using-devfile-lifecycle-events.md), like [`postStart`](../user-guides/advanced/using-devfile-lifecycle-events.md#poststart)
4. Synchronize the local files to the Dev environment
5. Start the port forwarding logic

> Note that this is the default behavior of `odo dev` if the Devfile does not define any commands at all.

Reviewer: ```suggestion
Note that this is the default behavior of `odo dev` if the Devfile does not define any *default* Build, Run or Debug commands.
```

Author: When there is only one command of a given kind, `odo` currently automatically uses it even if it is not explicitly marked as the default: https://github.com/redhat-developer/odo/blob/b6c9c882459535df27e78422b00c680918eae533/pkg/libdevfile/libdevfile.go#L103-L106

Do we want to change that behavior?

That's why I considered limiting this to the case where there are no commands at all, like with the [UDI](https://registry.devfile.io/viewer/devfiles/community/udi) Stack.

Author: Alright, that makes sense. I'll update it. Thanks.
